### PR TITLE
Bug 1308122 - Make it return 400 when PerformanceSignatureViewSet meet ValueError

### PR DESCRIPTION
**Pull request by MikeLing**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1976)

### CODE REVIEW

File: treeherder/webapp/api/performance_data.py [hunk 1]
```diff
@@ -46,12 +46,18 @@ def list(self, request, project):
                 parent_signature__in=parent_signatures)
 
         if not int(request.query_params.get('subtests', True)):
-            signature_data = signature_data.filter(parent_signature__isnull=True)
+            try:
+                signature_data = signature_data.filter(parent_signature__isnull=True)
+            except ValueError:
+                return "No available subtest provided", HTTP_400_BAD_REQUEST
```
Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: I don't think we need this, I don't see how we could get a ValueError exception here.

File: treeherder/webapp/api/performance_data.py [hunk 2]
```diff
@@ -46,12 +46,18 @@ def list(self, request, project):
...
 
         signature_ids = request.query_params.getlist('id')
         if signature_ids:
-            signature_data = signature_data.filter(id__in=map(int,
-                                                              signature_ids))
+            try:
+                signature_data = signature_data.filter(id__in=map(int,
+                                                                  signature_ids))
+            except ValueError:
+                return "No id provided", HTTP_400_BAD_REQUEST
```
Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: I think you need to return a Response object here, not a tuple.

Also, let's make the error message "One or more id values invalid (must be integer)"

File: treeherder/webapp/api/performance_data.py (hunk 1 above)
Comment (MikeLing) [inline treeherder/webapp/api/performance_data.py]: Ed said we may got ValueError if someone use "1 and 2" as subtests https://bugzilla.mozilla.org/show_bug.cgi?id=1308122#c0

**Comment by edmorley** [comment]
@MikeLing do you know if you'll get a chance to look at this? :-)

**Comment by edmorley** [comment]
Closing for inactivity

**Comment by wlach** [comment]
Actually as far as I can tell, @MikeLing actually fixed my issues and this is ready to merge! He just forgot to comment.

**Pull request merged**